### ISSUE TITLE
podman rmi <image-referenced-by-another>: removes as well as untags

### ISSUE THREAD

**Issue by edsantiago** [projectatomic/libpod#992]
This may be a placeholder for when `podman images --all` is implemented.
```
# podman run --name c_foo fedora true
# podman commit -q c_foo foo
2dc77d815ffff136e6c7d7c4986894139cef2b58da2df352addfa1e2ceb6bb15
# podman rm c_foo
0bbb1816555c795d14ed4ed329e05fe053e33cee277c394445462570d21c1294
# podman images -a
REPOSITORY                 TAG      IMAGE ID       CREATED         SIZE
localhost/foo              latest   2dc77d815fff   8 seconds ago   263MB
docker.io/library/fedora   latest   cc510acfcd70   7 weeks ago     263MB
# podman rmi fedora
cc510acfcd701a409014118d5f417f0022520802a26c650866b8a9594d75f3a7
# podman images -a
REPOSITORY      TAG      IMAGE ID       CREATED          SIZE
localhost/foo   latest   2dc77d815fff   16 seconds ago   263MB
```
Note that `rmi` didn't just untag fedora, it seems to have removed the image even though that image is referenced by `foo`. This affects image history:
```
# podman history foo
ID             CREATED          CREATED BY                                      SIZE      COMMENT
2dc77d815fff   37 seconds ago                                                   2.56kB    
<missing>      7 weeks ago      /bin/sh -c #(nop) ADD file:a09c7346f23d88d...   262.7MB   
<missing>      7 months ago     /bin/sh -c #(nop) ENV DISTTAG=f28container...   262.7MB   
<missing>      7 months ago     /bin/sh -c #(nop) MAINTAINER [Adam Miller ...   262.7MB   
```
(on line 2, I would expect to see `cc510acfc`). Compare to docker:
```
...
# docker rmi fedora
Untagged: fedora:latest
Untagged: docker.io/fedora@sha256:7ae08e5637170eb47c01e315b6e64e0d48c6200d2942c695d0bee61b38c65b39
     (^^^ untagged, not removed)
# docker images -a
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
foo                 latest              9675f7017482        30 seconds ago      253 MB
<none>              <none>              cc510acfcd70        7 weeks ago         253 MB
# docker history foo
IMAGE               CREATED             CREATED BY                                      SIZE                COMMENT
9675f7017482        47 seconds ago      true                                            0 B                 
cc510acfcd70        7 weeks ago         /bin/sh -c #(nop) ADD file:a09c7346f23d88d...   253 MB              
<missing>           7 months ago        /bin/sh -c #(nop)  ENV DISTTAG=f28containe...   0 B                 
<missing>           7 months ago        /bin/sh -c #(nop)  MAINTAINER [Adam Miller...   0 B  
# docker rmi cc510
Error response from daemon: conflict: unable to delete cc510acfcd70 (cannot be forced) - image has dependent child images
```
podman-0.6.4-1.gitd5beb2f.fc28

**Comment by rhatdan** [comment]
Urvashi, PTAL

**Comment by umohnani8** [comment]
Fixed in #1055
